File: rust/p0287_find_duplicate_number/src/brute_force.rs
pub struct Solution;

impl Solution {
    pub fn find_duplicate(nums: Vec<i32>) -> i32 {
        for i in 0..nums.len() {
            for j in i + 1..nums.len() {
                if nums[i] == nums[j] {
                    return nums[i];
                }
            }
        }
        unreachable!("Problem guarantees 1 duplicate and nums.length > 1")
    }
}

File: csharp/P0287_FindDuplicateNumber/Solutions/BruteForce.cs
namespace P0287_FindDuplicateNumber.BruteForce;

// LeetCode convention: the class is named "Solution"
public class Solution
{
    // Brute force implementation
    public int FindDuplicate(int[] nums)
    {
        for (int i = 0; i < nums.Length; i++)
        {
            for (int j = i + 1; j < nums.Length; j++)
            {
                if (nums[i] == nums[j])
                    return nums[i];
            }
        }

        // This should never happen
        throw new ArgumentException("Input array must contain at least one duplicate");
    }
}
